### PR TITLE
feat: add collector name validation in doctor

### PR DESCRIPTION
**Pull request by rahulguptajss**

### CODE REVIEW

**Review by Hardikl — DISMISSED** [review]

File: pkg/util/util.go [hunk 1]
```diff
@@ -20,6 +20,15 @@ import (
 	"strings"
 )
 
+var IsCollector = map[string]struct{}{
```
Comment (cgrinds) [inline pkg/util/util.go:23]: missing Unix and Simple

File: cmd/tools/doctor/doctor.go [hunk 2]
```diff
@@ -132,6 +135,43 @@ func checkAll(path string, contents []byte) {
 	}
 }
 
+// checkCollectorName checks if the collector names in the config struct are valid
+func checkCollectorName(config conf.HarvestConfig) validation {
+	valid := validation{isValid: true}
+
+	// Verify default collectors
+	defaultCollectors := config.Defaults.Collectors
+	for _, c := range defaultCollectors {
```
Comment (cgrinds) [inline cmd/tools/doctor/doctor.go:144]: panic when Defaults does not exist

File: cmd/tools/doctor/doctor.go [hunk 3]
```diff
@@ -132,6 +135,43 @@ func checkAll(path string, contents []byte) {
...
+		// Check if the collector name is valid
+		_, ok := util.IsCollector[c.Name]
+		if !ok {
+			fmt.Printf("Default Section uses an invalid collector %s \n", color.Colorize(c.Name, color.Red))
+			valid.isValid = false
+		}
+	}
+
+	// Verify poller collectors
+	for k, v := range config.Pollers {
```
Comment (cgrinds) [inline cmd/tools/doctor/doctor.go:154]: panic when Pollers does not exist

File: cmd/tools/doctor/doctor.go [hunk 4]
```diff
@@ -132,6 +135,43 @@ func checkAll(path string, contents []byte) {
...
+			if !ok {
+				fmt.Printf("Poller [%s] uses an invalid collector [%s] \n", color.Colorize(k, color.Yellow), color.Colorize(c.Name, color.Red))
+				valid.isValid = false
+			}
+		}
+	}
+
+	// Print the valid collector names if there are invalid collector names
+	if !valid.isValid {
+		keys := reflect.ValueOf(util.IsCollector).MapKeys()
```
Comment (cgrinds) [inline cmd/tools/doctor/doctor.go:168]: reflection isn't needed, better to idiomatic code

**Review by cgrinds — COMMENTED** [review]
please add a unit test to cover the comments

**Review by Hardikl — APPROVED** [review]

**Pull request merged**